### PR TITLE
feat(BoardDetails)[OK-185] Integrate board details view with backend

### PR DESCRIPTION
**Pull request by adbista**
This PR introduces changes in both permissions page and main board details page.

### CODE REVIEW

File: client/src/components/board-details/board-permissions/BoardPermissions.tsx [hunk 1]
```diff
@@ -11,38 +11,53 @@ import { Table, TableBody, TableCell, TableRow } from "@/components/ui/table";
...
-      permission: BoardPermission.EDITOR,
-      addedAt: "yesterday 10:42 AM",
-      invitedBy: "Cnon Cnna",
-    },
-  ]);
+import { BoardDetailsResponse } from "@/interfaces/responses/board-details-response";
+import logger from "@/lib/logger";
+export const BoardPermissions = ({ board }: { board: BoardDetailsResponse }) => {
+  const mapPermissions = useCallback((board: BoardDetailsResponse): BoardPermissionsUser[] => {
+    const { viewer, editor, moderator } = board.permissions;
```
Comment (Strattoss) [inline client/src/components/board-details/board-permissions/BoardPermissions.tsx]: `viewer`, `editor` and `moderator` are arrays, right? If so, I would change the name to something more like
```suggestion
    const { viewer: viewers, editor: editors, moderator: moderators } = board.permissions;
```

**Review by Strattoss — APPROVED** [review]
Nicee

File: client/src/components/board-details/board-permissions/BoardPermissions.tsx (hunk 1 above)
Comment (adbista) [reply client/src/components/board-details/board-permissions/BoardPermissions.tsx]: done

Comment (adbista) [reply client/src/components/board-details/board-permissions/BoardPermissions.tsx]: If we want to change that entirely, backend needs to make some changes also

**Pull request merged**